main/src/ca/uwaterloo/flix/api/lsp/provider/CodeLensProvider.scala

```diff
@@ -26,19 +26,15 @@ object CodeLensProvider {
   /**
     * Processes a codelens request.
     */
-  def processCodeLens(uri: String)(implicit index: Index, root: Root): JObject = {
+  def processCodeLens(uri: String)(implicit index: Index, root: Option[Root]): JObject = {
     val codeLenses = getRunCodeLenses(uri) ::: getTestCodeLenses(uri)
     ("status" -> "success") ~ ("result" -> JArray(codeLenses.map(_.toJSON)))
   }
 
   /**
     * Returns code lenses for running entry points.
     */
-  private def getRunCodeLenses(uri: String)(implicit index: Index, root: Root): List[CodeLens] = {
-    if (root == null) {
-      return Nil
-    }
-
+  private def getRunCodeLenses(uri: String)(implicit index: Index, root: Option[Root]): List[CodeLens] = {
```

Comment:
instead of this map and getOrElse business, I would just do a pattern match so the code looks similar:

root match {
  case None => Nil
  case Some(root) => ...